tests/helixpublishwitharcade.proj

```diff
@@ -34,51 +30,121 @@
       <_ProjectsToBuild Include="$(MSBuildProjectFile)">
         <AdditionalProperties>$(_PropertiesToPass);Scenario=%(_Scenarios.Identity)</AdditionalProperties>
       </_ProjectsToBuild>
     </ItemGroup>
 
     <PropertyGroup>
       <_BuildInParallel>false</_BuildInParallel>
-      <_BuildInParallel Condition=" '@(_ProjectsToBuild->Count())' > '1' ">true</_BuildInParallel>
+      <_BuildInParallel Condition=" '@(_ProjectsToBuild->Count())' &gt; '1' ">true</_BuildInParallel>
     </PropertyGroup>
 
-    <MSBuild Projects="@(_ProjectsToBuild)" Targets="CreateTestEnvFiles" StopOnFirstFailure="true" />
     <MSBuild Projects="@(_ProjectsToBuild)" Targets="Test" BuildInParallel="$(_BuildInParallel)" StopOnFirstFailure="false" Properties="UsesHelixSdk=true" />
   </Target>
 
-  <Import Project="..\dir.props" />
+  <Import Project="src\dir.props" />
 
   <PropertyGroup>
-    <CoreRootDir>$(TestWorkingDir)Tests\Core_Root</CoreRootDir>
+    <BinDir>$([MSBuild]::NormalizeDirectory($(BinDir)))</BinDir>
+    <CoreRootDirectory>$(BinDir)Tests\Core_Root\</CoreRootDirectory>
+    <PayloadsRootDirectory>$(BinDir)Payloads\</PayloadsRootDirectory>
     <TestEnvFileName Condition=" '$(TargetsWindows)' == 'true' ">SetStressModes_$(Scenario).cmd</TestEnvFileName>
     <TestEnvFileName Condition=" '$(TargetsWindows)' != 'true' ">SetStressModes_$(Scenario).sh</TestEnvFileName>
   </PropertyGroup>
 
-  <ItemGroup>
-    <XUnitWrapperDll Include="$(TestWorkingDir)\**\*.XUnitWrapper.dll">
-      <FileDirectory>%(RootDir)%(Directory)</FileDirectory>
-      <File>%(FileName)%(Extension)</File>
-    </XUnitWrapperDll>
-  </ItemGroup>
+  <Import Project="testgrouping.proj" />
 
-  <Target Name="CreateTestEnvFiles">
+  <Target Name="DiscoverAllXUnitWrappers">
+    <ItemGroup>
+      <_XUnitWrapperDll Include="%(TestGrouping.XUnitWrapperDll)" Condition="Exists('%(XUnitWrapperDll)')">
+        <PayloadGroup>%(TestGroup)</PayloadGroup>
+      </_XUnitWrapperDll>
+      <_XUnitWrapperDll Include="@(XUnitWrapperGrouping)" />
+
+      <!-- This adds the remaining *.XUnitWrapper.dll files in BinDir unless
+           1) they are in PayloadsRootDirectory or
+           2) they are grouped by XUnitWrapperGrouping. -->
+      <_XUnitWrapperDll Include="$(BinDir)**\*.XUnitWrapper.dll" Exclude="$(PayloadsRootDirectory)**\*.XUnitWrapper.dll;@(XUnitWrapperGrouping->Metadata('FullPath'))">
+         <PayloadGroup></PayloadGroup>
+      </_XUnitWrapperDll>
+
+      <!-- Assign "default" value of PayloadGroup metadata. -->
+      <_XUnitWrapperDll Update="@(_XUnitWrapperDll)">
+         <PayloadGroup Condition="'%(PayloadGroup)' == ''">$([MSBuild]::ValueOrDefault(%(FileName),'').Replace('.XUnitWrapper',''))</PayloadGroup>
+      </_XUnitWrapperDll>
+
+      <_XUnitWrapperDll Update="@(_XUnitWrapperDll)">
+        <FileDirectory>%(RootDir)%(Directory)</FileDirectory>
+        <File>%(FileName)%(Extension)</File>
```

Comment:
Do we need to add `FileDirectory` and `File` metadata? I'd prefer to expand these when they're used (as long as they aren't used often).

Reply:
I probably can change this, so I won't need them (they are used right now but I think refCount is 1)